src/app/clusters/identify-server/identify-server.cpp

```diff
@@ -183,13 +183,14 @@ static void onIdentifyClusterTick(chip::System::Layer * systemLayer, void * appS
         if (EMBER_ZCL_STATUS_SUCCESS == Clusters::Identify::Attributes::GetIdentifyTime(endpoint, &identifyTime) &&
             0 != identifyTime)
         {
+            identifyTime = static_cast<uint16_t>(identifyTime == 0 ? 0 : identifyTime - 1);
             // This tick writes the new attribute, which will trigger the Attribute
             // Changed callback below, which in turn will schedule or cancel the tick.
             // Because of this, the tick does not have to be scheduled here.
-            (void) Clusters::Identify::Attributes::SetIdentifyTime(endpoint,
-                                                                   static_cast<uint16_t>(identifyTime == 0 ? 0 : identifyTime - 1));
+            (void) Clusters::Identify::Attributes::SetIdentifyTime(endpoint, identifyTime);
         }
-        else
+
+        if (identifyTime == 0)
         {
             identify->mActive = false;
         }
```

Comment:
This won't work. The name is maybe not perfect but `scheduleIdentifyTick` does more than that. It would break the callback logic of `mOnIdentifyStop` and `mOnIdentifyStart`.

Reply:
Given the actual setting of `mActive` to 'true' happens in `scheduleIdentifyTick`, I think that's the right place to set it to false as well. That would consolidate all mutations to `mActive` in a single function flow.

Would that work @eve-cxrp ?

Reply:
I have now refactored the code. Feel free to check if this is better.